src/UI/Component/JSBindable.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace ILIAS\UI\Component;
+
+/**
+ * Interface to be extended by components that have the possibility to bind
+ * to Javascript.
+ */
+interface JSBindable {
+	/**
+ 	 * Add some JS-code that binds to the given component on load.
+	 *
+	 * The closure gets the id of the component as string and is expected to return a string
+	 * containing javascript statements.
+	 *
+	 * You should not assume anything about the id besides that it uniquely identifies the
+	 * DOM-Element that represents the component you have bound to in HTML. This means: the
+	 * id may change between different requests or on different pages. The id can not be
```

Comment:
Add some javascript via `ComponentRenderer::registerResource`. The common pattern atm seems to be to register some object in the JS global namespace, where the name resembles the component (e.g. found `ilDataCollection`). The object then has some methods to do further processing, right? On your component you could then do `$my_component->withOnLoadCode(function($id) { return "ilMyComponent.doSomeProcessing(\"$id\"); });`.